### ISSUE TITLE
v1.7.x: kube-proxy crash because of nil pointer

### ISSUE THREAD

**Issue by pires** [kubernetes/kubernetes#49780]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> /kind bug

**What happened**:

Ran `kube-proxy` 1.7.2 with the same configuration as previously working v1.6.7 and it crashed:
```
./kube-proxy --hostname-override=kube-node-0.kubernetes.mesos --master=https://apiserver.kubernetes.mesos:6443 --kubeconfig=kubeconfig-path/kubeconfig.yaml --healthz-port=0

W0728 12:45:05.444943      73 server.go:188] WARNING: all flags other than --config, --write-config-to, and --cleanup-iptables are deprecated. Please begin using a config file ASAP.
I0728 12:45:05.579883      73 server.go:474] Using iptables Proxier.
W0728 12:45:05.592373      73 server.go:783] Failed to retrieve node info: nodes "kube-node-0.kubernetes.mesos" not found
W0728 12:45:05.592742      73 proxier.go:483] invalid nodeIP, initializing kube-proxy with 127.0.0.1 as nodeIP
W0728 12:45:05.592764      73 proxier.go:488] clusterCIDR not specified, unable to distinguish between internal and external traffic
I0728 12:45:05.592850      73 server.go:509] Tearing down userspace rules.
I0728 12:45:05.762516      73 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I0728 12:45:05.762580      73 conntrack.go:52] Setting nf_conntrack_max to 131072
I0728 12:45:05.762631      73 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I0728 12:45:05.762651      73 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1abd1a6]

goroutine 1 [running]:
k8s.io/kubernetes/pkg/proxy/healthcheck.(*HealthzServer).UpdateTimestamp(0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/healthcheck/healthcheck.go:292 +0x26
k8s.io/kubernetes/pkg/proxy/iptables.(*Proxier).SyncLoop(0xc420198160)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/iptables/proxier.go:642 +0x62
k8s.io/kubernetes/cmd/kube-proxy/app.(*ProxyServer).Run(0xc420245320, 0xc420442000, 0xc42021d680)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:723 +0x493
k8s.io/kubernetes/cmd/kube-proxy/app.(*Options).Run(0xc420442000, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:226 +0xa0
k8s.io/kubernetes/cmd/kube-proxy/app.NewProxyCommand.func1(0xc420413b00, 0xc420134940, 0x0, 0x4)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:341 +0x80
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc420413b00, 0xc42000e100, 0x4, 0x4, 0xc420413b00, 0xc42000e100)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x22b
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420413b00, 0xc420681f60, 0xc420681f58, 0xc4202e9c20)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x339
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420413b00, 0x20470f0, 0xc4200001a0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/proxy.go:44 +0x96
```

**What you expected to happen**:

`kube-proxy` works as in version 1.6.7.

**How to reproduce it (as minimally and precisely as possible)**:

The command line is show in the first line of the log above.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.7.2 (all components from v1.7.2 tag)
- Cloud provider or hardware configuration: Running in Mesos similar to how kubernetes-mesos framework used to.

**Comment by ncdc** [comment]
This was recently fixed by #49137 in master.

**Comment by pires** [comment]
Thank you @ncdc. Raising awareness to get this backported to 1.7.

**Comment by cmluciano** [comment]
https://github.com/kubernetes/kubernetes/issues/49781

**Comment by RRAlex** [comment]
Does this also fix the issue of kube-api (1.7.2) crashing bellow?

```
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.351782       1 wrap.go:42] GET /apis/extensions/v1beta1/thirdpartyresources: (3.213013ms) 200 [[kube-apiserver/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:39508]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.684847       1 wrap.go:42] GET /api/v1/pods: (8.925452ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.705389       1 wrap.go:42] GET /api/v1/replicationcontrollers: (2.536395ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.709419       1 wrap.go:42] GET /api/v1/services: (3.465677ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.716461       1 wrap.go:42] GET /apis/apps/v1beta1/statefulsets: (1.638575ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.719147       1 wrap.go:42] GET /apis/autoscaling/v1/horizontalpodautoscalers: (2.145134ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.721666       1 wrap.go:42] GET /apis/batch/v1/jobs: (2.050715ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.724633       1 wrap.go:42] GET /apis/extensions/v1beta1/deployments: (2.556245ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.727730       1 wrap.go:42] GET /apis/extensions/v1beta1/replicasets: (1.90639ms) 200 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: I0729 17:50:33.731469       1 wrap.go:42] GET /apis/extensions/v1beta1/networkpolicies: (2.632383ms) 200
 [[kubectl/v1.7.2 (linux/amd64) kubernetes/922a86c] 127.0.0.1:58790]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: panic: runtime error: invalid memory address or nil pointer dereference
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8913dc]
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: 
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: goroutine 4715006 [running]:
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/vendor/golang.org/x/net/http2.strSliceContains(0x0, 0x1, 0x1, 0x3c1fe50, 0x2, 0x0)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/transport.go:2004 +0x3c
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/vendor/golang.org/x/net/http2.configureTransport(0xc4256c22d0, 0x0, 0x0, 0x3d3e300)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/configure_transport.go:28 +0x136
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/vendor/golang.org/x/net/http2.ConfigureTransport(0xc4256c22d0, 0xd, 0x0)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang.org/x/net/http2/transport.go:125 +0x2b
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/net.SetTransportDefaults(0xc4256c22d0, 0xc4256c22d0)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/net/http.go:86 +0xb2
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/pkg/probe/http.NewWithTLSConfig(0xc4292f8480, 0xc421581e08, 0xc421581e00)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/probe/http/http.go:40 +0x62
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/pkg/registry/core/componentstatus.(*Server).DoServerCheck.func1()
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/registry/core/componentstatus/validator.go:71 +0x3c
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: sync.(*Once).Do(0xc425b3c290, 0xc421581eb8)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/usr/local/go/src/sync/once.go:44 +0xbe
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/pkg/registry/core/componentstatus.(*Server).DoServerCheck(0xc425b3c240, 0xc42356ae80, 0xc42001ac00, 0x20003, 0xc421581f9e, 0xc421581f98, 0x0)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/registry/core/componentstatus/validator.go:72 +0x76
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/pkg/registry/core/componentstatus.(*REST).getComponentStatus(0xc420b24220, 0xc424a435e8, 0x6, 0xc425b3c240, 0x432668)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/registry/core/componentstatus/rest.go:98 +0x40
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: k8s.io/kubernetes/pkg/registry/core/componentstatus.(*REST).List.func1(0xc424a43610, 0xc420b24220, 0xc425b3c360, 0xc424a435e8, 0x6, 0xc425b3c240)
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/registry/core/componentstatus/rest.go:62 +0x73
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: created by k8s.io/kubernetes/pkg/registry/core/componentstatus.(*REST).List
Jul 29 13:50:33 localhost docker/kube-apiserver[1048]: #011/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/registry/core/componentstatus/rest.go:64 +0x164
```

**Comment by xiangpengzhao** [comment]
@RRAlex seems like not. Could you please open a new issue to track the crashing you mentioned?

**Comment by thockin** [comment]
Who is backporting?

**Comment by pires** [comment]
It's fixed in 1.7.3.
